src/repository.rs

```diff
@@ -420,6 +453,75 @@ impl<M: Model> Repository<M> {
         self.coll.update_one(query, update, options).await
     }
 
+    /// Apply multiple update operations in bulk.
+    ///
+    /// This will be removed once support for bulk update is added to the official driver.
+    /// [see](https://jira.mongodb.org/browse/RUST-531) for tracking progress on this feature in the official driver.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let bulk_update_res = repository
+    ///     .bulk_update(&vec![
+    ///         &BulkUpdate {
+    ///             query: doc! { f!(name in User): "Dane" },
+    ///             update: doc! { Set: { f!(age in User): 12 } },
+    ///             options: None,
+    ///         },
+    ///         &BulkUpdate {
+    ///             query: doc! { f!(name in User): "David" },
+    ///             update: doc! { Set: { f!(age in User): 30 } },
+    ///             options: None,
+    ///         },
+    ///     ])
+    ///     .await
+    ///     .unwrap();
+    /// assert_eq!(bulk_update_res.nb_affected, 2);
+    /// assert_eq!(bulk_update_res.nb_modified, 2);
+    /// ```
+    pub async fn bulk_update<V, U>(
+        &self,
+        updates: V,
```

Comment:
Note:
The reason the signature is with Borrow is that it allow the following form of usage:
```rust
repository
    .bulk_update(vec![
        BulkUpdate {
            query: doc! { f!(name in User): "Dane" },
            update: doc! { Set: { f!(age in User): 12 } },
            options: None,
        },
        BulkUpdate {
            query: doc! { f!(name in User): "David" },
            update: doc! { Set: { f!(age in User): 30 } },
            options: None,
        },
    ])
```
```rust
repository
    .bulk_update(vec![
        &BulkUpdate {
            query: doc! { f!(name in User): "Dane" },
            update: doc! { Set: { f!(age in User): 12 } },
            options: None,
        },
        &BulkUpdate {
            query: doc! { f!(name in User): "David" },
            update: doc! { Set: { f!(age in User): 30 } },
            options: None,
        },
    ])
```
```rust
repository
    .bulk_update(&vec![
        BulkUpdate {
            query: doc! { f!(name in User): "Dane" },
            update: doc! { Set: { f!(age in User): 12 } },
            options: None,
        },
        BulkUpdate {
            query: doc! { f!(name in User): "David" },
            update: doc! { Set: { f!(age in User): 30 } },
            options: None,
        },
    ])
```
```rust
repository
    .bulk_update(&vec![
        &BulkUpdate {
            query: doc! { f!(name in User): "Dane" },
            update: doc! { Set: { f!(age in User): 12 } },
            options: None,
        },
        &BulkUpdate {
            query: doc! { f!(name in User): "David" },
            update: doc! { Set: { f!(age in User): 30 } },
            options: None,
        },
    ])
```

Which can be convenient in some circumstances